src/main/java/net/kaczmarzyk/spring/data/jpa/web/SpecificationFactory.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright 2014-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.kaczmarzyk.spring.data.jpa.web;
+
+import net.kaczmarzyk.spring.data.jpa.utils.TypeUtil;
+import org.springframework.context.support.AbstractApplicationContext;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.lang.annotation.Annotation;
+import java.util.*;
+import java.util.function.Consumer;
+
+import static java.util.Objects.nonNull;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * @author Kacper Leśniak
```

Comment:
did you
1. write this class from scratch?
2. copied the preexisting logic from other classes?
3. copied the preexisting logic + added something new from you?

This author comment is correct only in the first case. (2) = original author or authors should be preserved (3) = you can add your name at the end of the list of original author or authors